### PR TITLE
Use build output dir for proof (fixes #222)

### LINKED ISSUE #222: Build Report shows proof for layer not assembled artifact
## Checklist

 - [X] Confirmed this is an issue with charm-tools, not charmstore-client
 - [ ] Provide versions of tools used
 - [ ] Described the feature or ways to replicate the issue

## What version am I running?

I ran the following command: `charm version` and got the following ouput:

```
charm unavailable
charm-tools 2.1.2
```

I am using: (Your operating system)

charmbox

## Issue/Feature
charm build -r shows output for layer not for assembled charm.

```shell
build: ---------------------------------------
build:               Build Report
build: ---------------------------------------
build: New build; all files were modified.
proof: I: metadata name (kubernetes) must match directory name (layer-k8s) exactly for local deployment.
proof: I: no hooks directory
proof: W: no copyright file
proof: I: all charms should provide at least one thing
proof: I: relation etcd has no hooks
proof: I: missing recommended hook install
proof: I: missing recommended hook start
proof: I: missing recommended hook stop
proof: I: missing recommended hook config-changed
```

### I expect/expected the following

```shell
I: relation dockerhost has no hooks
```

### PR DESCRIPTION
**Pull request by johnsca**
Description of change

## Checklist

 - [x] Have you followed [Juju Solutions hacking guide?](https://hacking.juju.solutions)
 - [x] Are all your commits [logically] grouped and squashed appropriately?
 - [x] Does this patch have code coverage?
 - [x] Does your code pass `make test`?

Fixes #222

### CODE REVIEW

File: charmtools/build/builder.py [hunk 1]
```diff
@@ -675,9 +675,14 @@ def main(args=None):
 
         build()
 
-        lint, exit_code = proof.proof(os.getcwd(), False, False)
+        lint, exit_code = proof.proof(build.output_dir, False, False)
```
Comment (chuckbutler) [inline charmtools/build/builder.py]: :clap: Perfect

**Comment by chuckbutler** [comment]
charm build -r
build: Composing into /home/ubuntu
build: Destination charm directory: /home/ubuntu/builds/filebeat
build: Processing layer: layer:basic
build: Processing layer: layer:apt
build: Processing layer: layer:beats-base
build: Processing layer: filebeat
build: Processing interface: elastic-beats
build: Processing interface: elasticsearch
build:
build: ---------------------------------------
build:               Build Report
build: ---------------------------------------
build: No new changes; no files were modified.

I didn't actually get the banner for proof, however the charm did pass proof cleanly. so LGTM 👍

**Comment by johnsca** [comment]
The banner only shows up if there is proof output.

**Comment by chuckbutler** [comment]
I suspected. This branch is +1 LGTM. are you cool if I hit the merge button? I wanted to leave this open for further review if required.

**Comment by marcoceppi** [comment]
can we get a rebase? I'd like to get this into the next patch release

**Comment by AdamIsrael** [comment]
This would be a really useful patch to land ahead of the charmer summit. I just ran into this running charm-tools installed via `snap install charm --edge`.

**Comment by marcoceppi** [comment]
Fixed in #256

**Pull request closed without merging**